mediatek: filogic: add Keenetic Sprinter (KN-3711) support

volatileobject opened the pull request:
Specification:
    SoC: Mediatek MT7981BA
    RAM: 512 MB DDR4 RAM
    Flash: Winbond W25N01GV 128 MB SPI-NAND (dual boot on OEM)
    WiFi: MT7976CN DBDC AX
    Switch: MediaTek MT7531AE (3x LAN Gigabit ports) + Internal Gbe Phy (1x WAN Gigabit port)
    GPIO: 4x LED (power, internet, fn, wifi), 2x buttons (wps, reset), 1x switch (mesh)

UART Interface:
    Pins: VCC, TX, RX, NC, GND
    Settings: 115200, 8N1

Flashing via TFTP:
    1. Connect your PC and router to the first LAN port, configure PC interface using IP 192.168.1.2, mask 255.255.255.0
    2. Save the factory image of the OpenWRT firmware, renamed to KN-3711_recovery.bin, on the TFTP server
    3. Hold the reset button and power up the device
    4. Keep the button pressed until the status LED starts blinking
    
Signed-off-by: Vladimir Epifantsev <volatilefield@outlook.com>

DragonBluep commented on target/linux/mediatek/dts/mt7981b-keenetic-kn-3711.dts:
```diff
@@ -0,0 +1,336 @@
+};
+
+&watchdog {
+	status = "okay";
```
This node can  be removed. The watchdog was already enabled by default in mt7981b.dtsi.

DragonBluep commented on target/linux/mediatek/dts/mt7981b-keenetic-kn-3711.dts:
```diff
@@ -0,0 +1,336 @@
+};
+
+&mdio_bus {
+	switch: switch@1f {
```
```suggestion
	switch@1f {
```

volatileobject commented on target/linux/mediatek/dts/mt7981b-keenetic-kn-3711.dts:
```diff
@@ -0,0 +1,336 @@
+
+		led-2 {
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&pio 9 GPIO_ACTIVE_LOW>;
```
Added.

volatileobject committed:
mediatek: filogic: add Keenetic Sprinter (KN-3711) support

Specification:
    SoC: Mediatek MT7981BA
    RAM: 512 MB DDR4 RAM
    Flash: Winbond W25N01GV 128 MB SPI-NAND (dual boot on OEM)
    WiFi: MT7976CN DBDC AX
    Switch: MediaTek MT7531AE (3x LAN Gigabit ports) + Internal Gbe Phy (1x WAN Gigabit port)
    GPIO: 4x LED (power, internet, fn, wifi), 2x buttons (wps, reset), 1x switch (mesh)

UART Interface:
    Pins: VCC, TX, RX, NC, GND
    Settings: 115200, 8N1

Flashing via TFTP:
    1. Connect your PC and router to the first LAN port, configure PC interface using IP 192.168.1.2, mask 255.255.255.0
    2. Save the factory image of the OpenWRT firmware, renamed to KN-3711_recovery.bin, on the TFTP server
    3. Hold the reset button and power up the device
    4. Keep the button pressed until the status LED starts blinking

Signed-off-by: Vladimir Epifantsev <volatilefield@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/19633
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

volatileobject commented on target/linux/mediatek/dts/mt7981b-keenetic-kn-3711.dts:
```diff
@@ -0,0 +1,333 @@
+					macaddr_factory_4: macaddr@4 {
+						compatible = "mac-base";
+						reg = <0x4 0x6>;
+						#nvmem-cell-cells = <1>;
```
I checked, it works correctly
